# Contents
1. [Introduction](#intro)<br>
    1.1. [Case Study ](#study)<br>
    1.2. [Objectives](#objs)
2. [Loading Libraries](#libs)
3. [Loading the Dataset](#datasets)
4. [Exploratory Data Analysis](#eda)<br>
    4.1. [Numerical EDA](#num-eda)<br>
    4.2. [Statistical EDA](#stats-eda)<br>
    4.2. [Visual EDA](#visual-eda)<br>

# 1.  Introduction
<a id="intro"></a>
Blood donation has been around for a long time. The first successful recorded transfusion was between two dogs in 1665, and the first medical use of human blood in a transfusion occurred in 1818. Even today, donated blood remains a critical resource during emergencies.
<br>
## Case Study: <a id="study"></a>
A mobile blood donation vehicle in Taiwan. The Blood Transfusion Service Center drives to different universities and collects blood as part of a blood drive.
<br>
## Objectives <a id="objs"></a>
1.  To predict whether or not a donor will give blood the next time the vehicle comes to campus. (March 2007)

# 2. Loading Libraries
<a id="libs"></a>

In [3]:
%matplotlib inline
import numpy as np
import pandas as pd
import seaborn as sns
import folium
import warnings

# warnings.filterwarnings("ignore")

# Set default seaborn styles
sns.set()

# 3. Loading the Dataset
<a id="datasets"></a>

In [4]:
donations_dataset = pd.read_csv("../data/transfusion.data.csv")
donations_dataset.head()

Recency (months)  Frequency (times)  Monetary (c.c. blood)  Time (months)  \
0                 2                 50                  12500             98   
1                 0                 13                   3250             28   
2                 1                 16                   4000             35   
3                 2                 20                   5000             45   
4                 1                 24                   6000             77   

   whether he/she donated blood in March 2007  
0                                           1  
1                                           1  
2                                           1  
3                                           1  
4                                           0

# 4. Exploratory Data analysis(EDA)
<a id="eda"></a>
## Numerical EDA
<a id="num-eda"></a>

In [5]:
# Data information
donations_dataset.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 748 entries, 0 to 747
Data columns (total 5 columns):
Recency (months)                              748 non-null int64
Frequency (times)                             748 non-null int64
Monetary (c.c. blood)                         748 non-null int64
Time (months)                                 748 non-null int64
whether he/she donated blood in March 2007    748 non-null int64
dtypes: int64(5)
memory usage: 29.3 KB


### Dataset Description
1. The dataset has a total of 748 entries(examples)
2. The dataset has a total of 5 columns, namely:<br>
    **Recency(Months)** - Number of months since last donations<br>
    **Frequency(Times)** - Total Number of Donations donations<br>
    **Monetary(c.c. blood)** - Total volume of blood donated in c.c.<br>
    **Time(Months)** - Number of months since first donations<br>
    **Ehether he/she donated blood in 2007**
3. The dataset has no null entries

## Statistical EDA
<a id="stats-eda"></a>

In [8]:
# Basic data Description
donations_dataset.describe()

Recency (months)  Frequency (times)  Monetary (c.c. blood)  \
count        748.000000         748.000000             748.000000   
mean           9.506684           5.514706            1378.676471   
std            8.095396           5.839307            1459.826781   
min            0.000000           1.000000             250.000000   
25%            2.750000           2.000000             500.000000   
50%            7.000000           4.000000            1000.000000   
75%           14.000000           7.000000            1750.000000   
max           74.000000          50.000000           12500.000000   

       Time (months)  whether he/she donated blood in March 2007  
count     748.000000                                  748.000000  
mean       34.282086                                    0.237968  
std        24.376714                                    0.426124  
min         2.000000                                    0.000000  
25%        16.000000                                    0.000000  
50%        28.000000                                    0.000000  
75%        50.000000                                    0.000000  
max        98.000000                                    1.000000

### basic dataset statistics
1. The average number of months taken by donors since last donation is **9.5 months**.
2. The average number of months taken by donors since first donation is **34.3 months**.
3. The average number of total donations is **6 donations**.
4. An avergae of **1378.68 c.c.** was donated
5. The maximum total blood volume donated by a single donor was **12500 c.c.** while the minimum was **250 c.c.**.

## Visual EDA
<a id="visual-eda"></a>